smithy-model/src/main/java/software/amazon/smithy/model/node/internal/NodeHandler.java

```diff
@@ -79,7 +79,12 @@ void endNumber(String string, SourceLocation location) {
         if (string.contains(".")) {
             value = new NumberNode(new BigDecimal(string), location);
         } else {
-            value = new NumberNode(Long.parseLong(string), location);
+            try {
+                value = new NumberNode(Long.parseLong(string), location);
+            } catch (NumberFormatException e) {
+                value = new NumberNode(new BigDecimal(string), location);
+            }
+
         }
```

Comment:
Shouldn't this have the same logic as loading from the IDL? Interesting that this always parsed as a big decimal. It looks like previously this would have failed on `1E-1` or something like that since it tries to parse a long. In any event complex numbers should be decimals like the idl parser tries to do.

Reply:
Fixed in a4dcceb.